From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hiteshkumar Patel <hiteshkumarg@nvidia.com>
Date: Thu, 16 Oct 2025 05:54:45 +0000
Subject: [PATCH 015/254] NVIDIA: SAUCE: mailbox: tegra-hsp: Do not clear
 interrupt routing at probe

HSP interrupts are shared across firmware & OS. Currently, if a
firmware configures interrupt routing before kernel boots,
the tegra-hsp driver can clear that routing if the same shared
interrupt is assigned to Linux. This disables the interrupts for
that firmware.

Bug 4774805

Change-Id: Icd819a31398ab53f8e915b9b2ef1ffd234972eb3
Signed-off-by: Hiteshkumar Patel <hiteshkumarg@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3483088
(cherry picked from commit cc660d12b9be50593220c6a66c6c1b19c065b1cc)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3470952
Reviewed-by: Kartik Rajput <kkartik@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Brad Griffis <bgriffis@nvidia.com>
(cherry picked from commit 41152afc729bfbb8f3a737f2aa7a1e8773c9fe58)

diff --git a/drivers/mailbox/tegra-hsp.c b/drivers/mailbox/tegra-hsp.c
index 4abcf77ae22b7..fda5dbf95604d 100644
--- a/drivers/mailbox/tegra-hsp.c
+++ b/drivers/mailbox/tegra-hsp.c
@@ -836,7 +836,6 @@ static int tegra_hsp_request_shared_irq(struct tegra_hsp *hsp)
 {
 	unsigned int i, irq = 0;
 	unsigned int default_si = hsp->num_si;
-	unsigned int value;
 	int err;
 
 	for (i = 0; i < hsp->num_si; i++) {
@@ -854,13 +853,6 @@ static int tegra_hsp_request_shared_irq(struct tegra_hsp *hsp)
 
 		hsp->shared_irqs[i].enabled = true;
 
-		value = tegra_hsp_readl(hsp, HSP_INT_IE(i));
-		if (value && !hsp->soc->virtualized) {
-			dev_warn(hsp->dev,
-				 "disabling interrupts for si: %d\n", i);
-			tegra_hsp_writel(hsp, 0, HSP_INT_IE(i));
-		}
-
 		/* Use first available interrupt as default. */
 		if (default_si == hsp->num_si)
 			default_si = i;
-- 
2.43.0

